homeassistant/components/geniushub/sensor.py

```diff
@@ -58,6 +61,32 @@ def name(self):
         """Return the name of the sensor."""
         return self._name
 
+    @property
+    def icon(self):
+        """Return the icon of the sensor."""
+        values = self._device._info_raw['childValues']  # noqa; pylint: disable=protected-access
+
+        last_comms = utc_from_timestamp(values['lastComms']['val'])
+        interval = timedelta(seconds=values['WakeUp_Interval']['val'] * 3)
+
+        if last_comms < utcnow() - interval:
+            return "mdi:battery-unknown"
+
+        level = self.state
+
+        if level > 95:
+            return "mdi:battery"
+        if level > 85:
+            return "mdi:battery-80"
+        if level > 70:
+            return "mdi:battery-60"
+        if level > 55:
+            return "mdi:battery-40"
+        if level > 45:
+            return "mdi:battery-20"
```

Comment:
We have this helper:
https://github.com/home-assistant/home-assistant/blob/1e149a704bb38866828920e186190d7d2e7d0586/homeassistant/helpers/icon.py#L5-L20

The levels aren't exactly the same. Does it matter?

Reply:
The thing is, for (these z-wave devices), the vendor considers the battery level to be critical when it is <45% (rather than <5%), so was trying to reflect that.

I had thought about a similar function to above, so may mess with it to make it more elegant.